### PR TITLE
feat(Modal): add pop from center/anchor animations

### PR DESCRIPTION
**Pull request by YossiSaadi**
https://monday.monday.com/boards/3532714909/pulses/7586333722
https://monday.monday.com/boards/3532714909/pulses/7586334945

### CODE REVIEW

File: packages/core/src/components/ModalNew/utils/animationVariants.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    }
+  }
+};
+
+export const modalAnimationAnchorPopVariants: Variants = {
+  enter: {
+    opacity: [0, 1, 1],
+    scale: [0.65, 0.65, 1],
+    top: "50%",
+    left: "50%",
```
Comment (rivka-ungar) [inline packages/core/src/components/ModalNew/utils/animationVariants.ts:51]: Are the `top` & `left` needed here as they are defined already in the css?

File: packages/core/src/components/ModalNew/Modal/Modal.tsx [hunk 2]
```diff
@@ -67,41 +74,54 @@ const Modal = forwardRef(
...
-          </FocusLock>
-        </RemoveScroll>
-      </ModalProvider>
+              <FocusLock returnFocus>
+                <motion.div
+                  variants={modalAnimationVariants}
+                  initial="exit"
+                  animate="enter"
+                  exit="exit"
+                  custom={anchorElementRef}
```
Comment (rivka-ungar) [inline packages/core/src/components/ModalNew/Modal/Modal.tsx:102]: Does this need a fallback in case it is undefined, so it won't mess up the animation?

**Review by rivka-ungar — APPROVED** [review]
Very cool! Had 2 small questions

File: packages/core/src/components/ModalNew/Modal/Modal.tsx (hunk 2 above)
Comment (YossiSaadi) [reply packages/core/src/components/ModalNew/Modal/Modal.tsx:102]: checked it, if it is undefined then the pop from center variation would be applied and it doesn't use it
if it is being used from `custom`, then the pop from anchor variation would be applied, and it means that the anchor element ref isn't undefined

**Pull request merged**